libs/datasets/timeseries.py

```diff
@@ -398,6 +399,20 @@ def timeseries(self) -> pd.DataFrame:
                 dtype="float",
             )
 
+    def location_ids(self) -> pd.Index:
+        return self.static.index.unique(CommonFields.LOCATION_ID).union(
+            self.timeseries_bucketed.index.unique(CommonFields.LOCATION_ID)
+        )
+
+    @cached_property
+    def geo_data(self) -> pd.DataFrame:
+        location_ids = self.location_ids()
+        geo_data = dataset_utils.get_geo_data()
+        missing_location_id = location_ids.difference(geo_data.index)
+        if not missing_location_id.empty:
+            raise KeyError(f"location_id not in data/geo-data.csv:\n{missing_location_id}")
+        return dataset_utils.get_geo_data().reindex(index=location_ids)
```

Comment:
the `get_geo_data` function is called twice here, probably should make it a variable?

Reply:
Done